pkg/clusterlink/controllers/node/node_controller.go

```diff
@@ -171,10 +171,10 @@ func (r *Reconciler) CleanOrphan() error {
 	}
 	var errs []error
 	cnt := 0
-	for _, cn := range clusterNodes.Items {
-		clusterNode := cn
-		if _, ok := k8sNodeNameSet[cn.Spec.NodeName]; !ok {
-			if err := r.Delete(context.Background(), &clusterNode); err != nil {
+	for i := range clusterNodes.Items {
```

Comment:
why not  `for i, clusterNode := range ***`

Reply:
for _, clusterNode := range ****  will make a extra object copy from range to clusterNode

Reply:
> for _, clusterNode := range **** will make a extra object copy from range xxx to clusterNode

clusterNode  is a address. no copy.

Reply:
> > for _, clusterNode := range **** will make a extra object copy from range xxx to clusterNode
> 
> clusterNode is an address. no copy.

the declare is `[]ClusterNode `, it will make a copy, the no copy situation is `[]*clusterNode`

Reply:
But you use `clusterNodes.Items[i]` 
What is the differences

Reply:
Right, it should be `clusterNode := &clusterNodes.Items[I]`, I'll recommit it